Core/Core/Views/PublishedIconView.swift

```diff
@@ -0,0 +1,31 @@
+//
+// Copyright (C) 2019-present Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 3 of the License.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+import Foundation
+
+public class PublishedIconView: IconView {
+    public var published: Bool? {
+        didSet {
+            if published == true {
+                image = .icon(.publish)
+                tintColor = UIColor.named(.backgroundSuccess).ensureContrast(against: .white)
```

Comment:
Our named colors probably don't need the `ensureContrast` calls. That's mostly for customer-supplied colors. But it also probably doesn't hurt. 🤷‍♂️

Reply:
Oh good call. I'd rather not have them if I don't need them. Removed.